src/config/config.service.ts

```diff
@@ -27,6 +32,34 @@ export class ConfigService implements OnModuleInit {
       folders.shift();
       this.logger.warn(`No configuration found in ${folders.join(', ')}`);
     }
+
+    this.validateConfig(c);
+  }
+
+  /**
+   * Validates the union of default and user configuration parameters. The parser will
+   * not abort on first error but will highlight all errors detected. Validation is done
+   * strictly without conversion of types. An exception is raised on validation failure.
+   *
+   * @param cfg - configuration object to validate
+   */
+  // TODO Remove need for lint skip - need inspiration from the type gods
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  validateConfig(cfg: Object): void {
```

Comment:
I think one way to not use the object type would be using the `IConfig` type from the `config` package - just as an idea

Reply:
Just seen this ... I swear I tried every combination 8) Thanks.